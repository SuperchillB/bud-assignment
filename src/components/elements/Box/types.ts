import {
  BorderProps,
  ColorProps,
  FlexboxProps,
  LayoutProps,
  PositionProps,
  SpaceProps,
} from 'styled-system';

type BoxProps = SpaceProps &
  ColorProps &
  LayoutProps &
  FlexboxProps &
  BorderProps &
  PositionProps;

export default BoxProps;
