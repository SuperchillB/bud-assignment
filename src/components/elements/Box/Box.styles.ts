import styled from 'styled-components';
import {
  compose,
  space,
  layout,
  flexbox,
  border,
  position,
  color,
} from 'styled-system';
import BoxProps from './types';

export const Box = styled.div<BoxProps>(
  {
    boxSizing: 'border-box',
    minWidth: 0,
  },
  compose(space, color, layout, flexbox, border, position),
);
