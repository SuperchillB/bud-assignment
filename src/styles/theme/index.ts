import radii from './foundations/border';
import breakpoints from './foundations/breakpoints';
import { colors } from './foundations/colors';
import shadows from './foundations/shadows';
import space from './foundations/space';
import { fonts, fontSizes, fontWeights } from './foundations/typography';
import { ThemeType } from './types';

const theme: ThemeType = {
  breakpoints,
  colors,
  fonts,
  fontSizes,
  fontWeights,
  radii,
  shadows,
  space,
};

export default theme;
