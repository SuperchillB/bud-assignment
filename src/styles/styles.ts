import { createGlobalStyle } from 'styled-components';
import { normalize } from 'styled-normalize';
import { ThemeType } from './theme/types';

export const GlobalStyles = createGlobalStyle<{ theme: ThemeType }>`
  ${normalize};
  *, :before, :after {
    box-sizing: border-box;
    margin: 0;
    padding: 0;
  }
  html {
    font-size: 62.5%;
    scroll-behavior: smooth;
  }
  body {
    /* font-family: ${(props) => props.theme.fonts.primary}; */
    font-family: ${({ theme }) => theme.fonts.primary};
    font-size: 1.6rem;
    background: ${({ theme }) => theme.colors.info};
    color: ${(props) => props.theme.colors.primary}; // ! CHANGE LATER
    cursor: default;
  }
  main {
    height: 100%;
  }
  h1,h2,h3,h4,h5,h6,button {
    font-family: ${(props) => props.theme.fonts.title}; // ! CHANGE LATER
    line-height: 2rem;
  }
  p {
    line-height: 2rem;
  }
  h1 {
    font-weight: normal;
    font-size: 3.2rem;
  }
  a {
    color: inherit;
    text-decoration: none;
  }
  button {
    border: 0;
    padding: 0;
    background: none;
    cursor: pointer;
  }
  li{
    list-style: none;
  }
`;
