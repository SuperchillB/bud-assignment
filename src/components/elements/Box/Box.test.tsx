import { render } from '@testing-library/react';
import { Box } from './Box.styles';

describe('Box component', () => {
  test('renders with given props', () => {
    const { container } = render(
      <Box
        backgroundColor="budLightBlue"
        color="primary"
        mb={['medium', 'large', 'xlarge']}
        px="small"
      />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
