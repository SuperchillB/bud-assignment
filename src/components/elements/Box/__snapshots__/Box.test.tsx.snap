// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box component renders with given props 1`] = `
<div
  class="sc-bczRLJ gNKbKD"
  color="midnight"
/>
`;
